### PR TITLE
Data factory impl

### PR DESCRIPTION
**Pull request by ao508**
Briefly describe changes proposed in this pull request:
- Refactoring data parsing from JSON strings/java objects

--

## Crossing T's and dotting I's

Please follow these checklists to help prevent any unexpected issues from being introduced by the changes in this pull request. If an item does not apply then indicate so by surrounding the line item with `~~` to strikethrough the text. See [basic writing and formatting syntax](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for more information.

### Web service and data model checklist

- [n/a] ~~Have screenshots been uploaded to demonstrate changes made to the response body JSON schema and/or swagger page?~~
- [n/a] ~~Were updates made to the mocked incoming request data and/or mocked published request data?~~
- [n/a] ~~Have unit tests been updated in relation to updates to the mocked test data?~~

### Neo4j models and database schema checklist:
- [n/a] ~~Were any of the neo4j persistence models changed?~~
- [n/a] ~~Does the graph database produce the expected changes to models, relationships, and/or property names? [provide screenshot of updated elements in graph db]~~

### Message handlers checklist:
- [n/a] ~~Were changes introduced in this PR that may affect the workflow of incoming messages?~~
- [n/a] ~~Are messages following the expected workflow when published to the topic(s) changed in this pull request?~~
- [n/a] ~~Were unit tests added to ensure messages are handled as expected?~~

If no unit tests were updated or added, then please explain why: Refactoring code, no new logic was added.

### Configuration and/or permissions checklist:
- [n/a] ~~Were any new topics introduced?~~
- [n/a] ~~Were the topics and appropriate permissions updated in [cmo-metadb-configuration]~~(https://github.mskcc.org/cmo/cmo-metadb-configuration)?
- [n/a] ~~If a new account was set up then were the account credentials and keys committed to [cmo-metadb-configuration](https://github.mskcc.org/cmo/cmo-metadb-configuration)?~~
- [n/a] ~~Were the account credentials and keys shared with the appropriate parties?~~

### General checklist:
- [x] Have tests or has a separate issue that describes the types of tests that should be created been linked or created? If no test is included it should explicitly be mentioned in the PR why there is no test.
- [x] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)

### CODE REVIEW

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 1]
```diff
@@ -221,13 +224,10 @@ public void run() {
...
                         if (existingSample == null) {
-                            LOG.info("Sample metadata does not already exist - persting to db: "
+                            LOG.info("Sample metadata does not already exist - persisting to db: "
                                     + sampleMetadata.getPrimaryId());
                             // handle and persist new sample received
-                            MetadbSample sample = new MetadbSample();
-                            sampleMetadata.setImportDate(
-                                    LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE));
-                            sample.addSampleMetadata(sampleMetadata);
+                            MetadbSample sample = SampleDataFactory.buildNewResearchSample(sampleMetadata);
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:230]: should we rename this to buildNewResearchSampleFromMetadata ?

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 2]
```diff
@@ -367,13 +367,7 @@ public void onMessage(Message msg, Object message) {
...
                             new String(msg.getData(), StandardCharsets.UTF_8),
                             String.class);
-                    MetadbRequest request = mapper.readValue(requestJson,
-                            MetadbRequest.class);
-                    request.setRequestJson(requestJson);
-                    request.setMetaDbSampleList(extractMetadbSamplesFromIgoResponse(requestJson));
-                    request.setNamespace("igo");
-                    // creates and inits request metadata
-                    request.addRequestMetadata(extractRequestMetadata(requestJson));
+                    MetadbRequest request = RequestDataFactory.buildNewLimsRequest(requestJson);
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:370]: same here, change it to buildNewLimsRequestFromJson?

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 3]
```diff
@@ -392,12 +386,8 @@ public void onMessage(Message msg, Object message) {
...
                     String requestMetadataJson = mapper.readValue(
                             new String(msg.getData(), StandardCharsets.UTF_8), String.class);
-                    Map<String, String> requestMetadataMap =
-                            mapper.readValue(requestMetadataJson, Map.class);
-                    RequestMetadata requestMetadata = new RequestMetadata(
-                            requestMetadataMap.get("requestId"),
-                            requestMetadataJson,
-                            LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE));
+                    RequestMetadata requestMetadata =
+                            RequestDataFactory.buildNewRequestMetadata(requestMetadataJson);
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:390]: same here, to keep the naming consistent

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java [hunk 4]
```diff
@@ -40,6 +41,7 @@
     public MetadbSample saveSampleMetadata(MetadbSample
             sample) throws Exception {
         fetchAndLoadSampleDetails(sample);
+        //MetadbSample sample = dataFactory.setMetadbSamplePatientNode(metadbSample);
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java:44]: Needs to removed

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java [hunk 5]
```diff
@@ -107,13 +103,8 @@ public MetadbSample getMetadbSampleByRequestAndAlias(String requestId, SampleAli
...
         }
-        sample.setSampleMetadataList(sampleRepository
-                .findSampleMetadataListBySampleId(sample.getMetaDbSampleId()));
-        String cmoPatientId = sample.getLatestSampleMetadata().getCmoPatientId();
-        MetadbPatient patient = patientService.getPatientByCmoPatientId(cmoPatientId);
-        sample.setPatient(patient);
-        sample.setSampleAliases(sampleRepository.findAllSampleAliases(sample.getMetaDbSampleId()));
-        return sample;
+        return getAllMetadbSampleFields(sample);
+
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java:107]: extra line

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java [hunk 6]
```diff
@@ -123,13 +114,8 @@ public MetadbSample getMetadbSampleByRequestAndIgoId(String requestId, String ig
...
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java:118]: extra line

**Review by divyamadala30 — COMMENTED** [review]
A couple of tiny changes!

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java (hunk 1 above)
Comment (ao508) [reply service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:230]: good idea

**Pull request merged**